### DISCUSSION TITLE
Xdrip not picking up readings from miaomiao 3

### DISCUSSION THREAD

**Discussion by JimKnees** [NightscoutFoundation/xDrip#2967] [Q&A] [answered]
Hello,
I've been having issues with my xdrip+ all day since I changed my libre 2 this morning. I haven't been able to get xdrip to pick up the readings from my miaomiao 3. I have tried every fix I could find online and even isolated the issue to xdrip itself as glimp and tomato accept the readings just fine. Neither of these apps have the features of xdrip that I rely on so I'd appreciate any help anyone could offer. I'll attach a screenshot of my log to show the errors I've been getting. If you need any more information please just let me know. Any help is appreciated. Thank you. 

> 
![Screenshot_2023-07-26-01-43-49-12_2dd2c1bcb8f659cd696bcba2072f751b](https://github.com/NightscoutFoundation/xDrip/assets/140568131/df4fa783-8de2-4bbb-b193-2261ccf9f9ae)

**Comment by Navid200** [comment]
@PhilTU

**Reply by JimKnees** [reply]
Oh, I should have stated, my hardware is an oppo find X5 pro running android 13 and I'm on the most recent nightly as of 25/07

**Comment by JimKnees** [comment, marked as answer]
Hello. Exciting news! I fixed the issue after hours of trawling the web. My oop2 algorithm wasn't working. I installed the more recent oop2_more and everything started working again. Hope this might help someone in the future. Will be marking it as solved, or rather answered. Hope everyone has a good day/night or whatever.

**Comment by PhilTU** [comment]
Looks like another instance of the new C5 sensor which requires the latest OOP2 to recognise it and a version of xDrip post 23/5/23 for compatibility. Good work!
